### PR TITLE
Add developer guidelines

### LINKED ISSUE #59: Developer experience is not documented
Stuff like

- How to contribute to this code?
- How to run tests?
- How to run the formatter?

### PR DESCRIPTION
**Pull request by Sbozzolo**
And change formatter guidelines.

Closes #59

### CODE REVIEW

File: .github/workflows/JuliaFormatter.yml [hunk 1]
```diff
@@ -16,11 +16,6 @@ jobs:
     runs-on: ubuntu-latest
     timeout-minutes: 30
     steps:
-    - name: Cancel Previous Runs
```
Comment (juliasloan25) [inline .github/workflows/JuliaFormatter.yml:19]: why did you remove this?

File: README.md [hunk 2]
```diff
@@ -24,6 +24,139 @@ Shared utilities for packages within the CliMA project. Please, check out the
...
+### Tests and environments
+
+We prioritize well-tested code to guarantee `ClimaUtilities.jl` functions
+reliably. Here are some principles we follow:
+
+#### Tests are collected in the `test` folder and are exclusively there
+
+This means that:
+- All the tests can be run with `Pkg.test()` or `julia --project=test tests/runtests.jl`;
+- There are there are no special tests in the Buildkite pipeline;
```
Comment (juliasloan25) [inline README.md:51]: typo: double "there are"

File: README.md [hunk 3]
```diff
@@ -24,6 +24,139 @@ Shared utilities for packages within the CliMA project. Please, check out the
...
+- In fact, we use Buildkite only to run the tests with MPI/GPUs.
+
+#### There are no checked `Manifest.toml` files
```
Comment (juliasloan25) [inline README.md:54]: typo: "checked" -> "checked-in"

File: README.md [hunk 4]
```diff
@@ -24,6 +24,139 @@ Shared utilities for packages within the CliMA project. Please, check out the
...
+used by our Buildkite pipeline.
+
+> :note: Please, open an issue if you find workflow problems/friction with this
+> system.
+
+#### Running tests
+
+There are two equivalent ways to run tests. 
+
+First, Start a Julia session in the `ClimaDignostics` directory:
```
Comment (juliasloan25) [inline README.md:80]: typo: ClimaDignostics -> ClimaUtilities

File: README.md [hunk 5]
```diff
@@ -24,6 +24,139 @@ Shared utilities for packages within the CliMA project. Please, check out the
...
+```
+or use it from within the `TestEnv` or the `.buildkite` environments (see previous section).
+
+Then, you can format the package running:
+``` julia
+using JuliaFormatter; format(".")
+```
+or just with `format(".")` if the package is already imported.
+
+The rules for formatting are defined in the `.JuliaFormatter.toml`.
```
Comment (juliasloan25) [inline README.md:112]: where is this file? I don't see it in the top-level of ClimaUtilities, which is where I would expect it based on how this is written

**Review by juliasloan25 — APPROVED** [review]

File: .github/workflows/JuliaFormatter.yml (hunk 1 above)
Comment (Sbozzolo) [reply .github/workflows/JuliaFormatter.yml:19]: Because I added the concurrency group, which does the same thing but is built-in.

File: README.md (hunk 2 above)
Comment (Sbozzolo) [reply README.md:51]: Thank you!

File: README.md (hunk 3 above)
Comment (Sbozzolo) [reply README.md:54]: Thank youo!

File: README.md (hunk 5 above)
Comment (Sbozzolo) [reply README.md:112]: Thank you!

File: .github/workflows/JuliaFormatter.yml (hunk 1 above)
Comment (Sbozzolo) [reply .github/workflows/JuliaFormatter.yml:19]: Actually, the concurrency group was already there. So, this was redundant.
